src/ast/ast_validations.c

```diff
@@ -73,6 +77,29 @@ static void _AST_GetWithAliases(const cypher_astnode_t *node, TrieMap *aliases)
 	}
 }
 
+// Extract identifiers / aliases from a procedure call.
+static void _AST_GetProcCallAliases(const cypher_astnode_t *node, TrieMap *identifiers) {
+	// CALL db.labels() yield label
+	// CALL db.labels() yield label as l
+	assert(node && identifiers);
+	assert(cypher_astnode_type(node) == CYPHER_AST_CALL);
+
+	uint projection_count = cypher_ast_call_nprojections(node);
+	for(uint i = 0; i < projection_count; i++) {
+		const cypher_astnode_t *proj = cypher_ast_call_get_projection(node, i);
+		const cypher_astnode_t *alias_node = cypher_ast_projection_get_alias(proj);
+
+		if(alias_node) {
+			// Alias is given: YIELD label AS l.
+			const char *alias = cypher_ast_identifier_get_name(alias_node);
+			TrieMap_Add(identifiers, (char *)alias, strlen(alias), NULL, TrieMap_DONT_CARE_REPLACE);
+		} else {
+			// No alias use identifier as is: Alias is given: YIELD label
+			_CollectIdentifiers(proj, identifiers);
```

Comment:
I think that if the YIELD projection is unaliased, `proj` must be of type identifier, and we can just retrieve it as on 94 instead of making this call.

Reply:
It is of type `projection` is there a quick way to get a string representation of a projection node?

Reply:
You need to retrieve what is being projected (`cypher_ast_projection_get_expression(proj)`) first, but it seems like here the result of that call should be an `identifier` node just like an alias.

Reply:
👍